src/API.jl

```diff
@@ -348,7 +348,18 @@ function driverconnect(connstr)
     return dbc
 end
 
-connect(dsn, user, pwd) = driverconnect(string("DSN=", dsn, user === nothing ? "" : ";UID=$user", pwd === nothing ? "" : ";PWD=$pwd"))
+connect(dsn, extraauth) = driverconnect("$dsn;$extraauth")
+
+# should this be somewhere else? utils?
+# also question on escaping: apparently e.g. PWD={$pwd} is the correct way to escape a parameter with semicolons in it.
+# however there seems no standard way to escape PWD=mypass}hasacurly. (SQL server wants double }}, but other drivers differ).
+function getextraauth(usr::Union{AbstractString, Nothing}, pwd::Union{AbstractString, Nothing}, extraauth::Union{AbstractString, Nothing}) :: String
```

Comment:
I would move this definition to `src/dbinterface.jl` below the `Connection` method, since that's the only place it's used.